pkg/executor/sortexec/parallel_sort_spill_test.go

```diff
@@ -143,3 +144,19 @@ func TestParallelSortSpillDiskFailpoint(t *testing.T) {
 		failpointDataInMemoryThenSpillTest(t, ctx, exe, sortCase, dataSource)
 	}
 }
+
+func TestIssue55344(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("CREATE TABLE t0(c0 BOOL);")
+	tk.MustExec("INSERT INTO mysql.opt_rule_blacklist VALUES('predicate_push_down'),('column_prune');")
+	tk.MustExec("ADMIN reload opt_rule_blacklist;")
+
+	// Should not be panic
+	tk.MustQuery("SELECT t0.c0 FROM t0 WHERE 0 ORDER BY -646041453 ASC;")
+
+	tk.MustExec("delete from mysql.opt_rule_blacklist;")
```

Comment:
should only delete `predicate_pushd_down` and `column_prune`

Reply:
> should only delete `predicate_pushd_down` and `column_prune`

done